### ISSUE TITLE
Episode thumbnail is always downloaded even if not requested

### ISSUE THREAD

**Issue by psonnosp** [Komet/MediaElch#969]
**Describe the bug**
If you scrape an already scraped episode with a thumbnail and uncheck to scrape for thumbnail ME downloads a thumb anyway.

**To Reproduce**
1. Go to an episode that's already scraped with a thumbnail.
2. Ctrl-F and make sure 'Thumbnail' is unchecked and scrape the episode.
3. ME loads a thumbnail anyway.

**Expected behavior**
That ME respect the setting and leave the already existing thumbnail alone.

**MediaElch Version:**
 - [X] 2.6.7-dev (nightly)
 - [ ] 2.6.6 stable

**Operating System:**
 - [X] Windows
 - [ ] macOS
 - [ ] Linux (distribution: _._)

**Comment by psonnosp** [comment]
Decided to install an older version to check when this was introduced since this is something I've been doing a lot lately but never had it happen before.
But now it was present in all the seven different versions available for download and that shouldn't be possible since I've used them before doing this exact same thing without any problems.

I don't know where this is coming from, it's super weird and I don't even have my beer goggles on this time 😉

**Comment by psonnosp** [comment]
Figured it out.
Or at least what's happening, I'm leaving the part on how to fix it to you 😉 

By accident/luck/coincidence the episodes I happen to try this on to confirm something fishy was going on all had very(?) old NFOs with a <thumb> tag in them pointing to the thumb at TheTVDB. So ME loads that one and completely ignored the already present thumbnail and the setting to not scrape one for some reason.

**Comment by bugwelle** [comment]
> If you scrape an already scraped episode with a thumbnail and uncheck to scrape for thumbnail ME downloads a thumb anyway.

Confirmed. Also found at least one reason why this happens. In our code we have this section:

```cpp
    if (m_episodeInfosToLoad.contains(TvShowScraperInfos::Tags) && !loadedEpisode->thumbnail().isEmpty()) {
        episode->setThumbnail(loadedEpisode->thumbnail());
    }
```

We check if tags were required but instead of setting the tags, we set the thumbnail... :facepalm: 

I'll look into this.

**Comment by bugwelle** [comment]
Your issue is caused by these lines:

https://github.com/Komet/MediaElch/blob/a53a2deb5a1514543b80d7cdcd18f291d0ddef21/src/ui/tv_show/TvShowWidgetEpisode.cpp#L668-L676


You are right. For some reason MediaElch loads the episode thumbnail again if the URL is set. Which it is always if the URL exists in the NFO...

Thank you for debugging it. I'll look into it.

**Comment by bugwelle** [comment]
Ok, currently no work around. I'm working on a rewrite of the TV scrapers. This will likely be a part of it. :)

**Comment by bugwelle** [comment]
Should be fixed by #1133
